sds_in_a_box/sds_in_a_box_stack.py

```diff
@@ -85,20 +97,48 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
-                                            "OS_INDEX": "metadata"
-                                            }
+                                            "OS_INDEX": "metadata",
+                                            "S3_BUCKET": data_bucket.s3_url_for_object()},
+                                            layers=[lambda_alpha_.PythonLayerVersion(self, "SDSIndexerCodeLayer", entry=os.path.join(os.path.dirname(os.path.realpath(__file__)), "SDSCode/"))]
                                           )
 
         indexer_lambda.add_event_source(S3EventSource(data_bucket,
                                                       events=[s3.EventType.OBJECT_CREATED]
                                                       )
                                         )
         indexer_lambda.apply_removal_policy(cdk.RemovalPolicy.DESTROY)
+
+        # The purpose of this lambda function is to trigger off of a lambda URL.
+        query_api_lambda = lambda_.Function(self,
+                                          id="QueryAPILambda",
+                                          code=lambda_.Code.from_asset(os.path.join(os.path.dirname(os.path.realpath(__file__)), "SDSCode/")),
+                                          handler="queries.lambda_handler",
+                                          role=lambda_role,
+                                          runtime=lambda_.Runtime.PYTHON_3_7,
+                                          timeout=cdk.Duration.minutes(15),
+                                          memory_size=1000,
+                                          environment={
+                                            "OS_ADMIN_USERNAME": "master-user", 
+                                            "OS_ADMIN_PASSWORD_LOCATION": os_secret.secret_value.unsafe_unwrap(),
+                                            "OS_DOMAIN": domain.domain_endpoint,
+                                            "OS_PORT": "443",
+                                            "OS_INDEX": "metadata"
+                                            }, 
+                                            layers=[lambda_alpha_.PythonLayerVersion(self, "SDSQueryCodeLayer", entry=os.path.join(os.path.dirname(os.path.realpath(__file__)), "SDSCode/"))]
+                                          )
+        # add function url for lambda query API
+        lambda_query_api_function_url = lambda_.FunctionUrl(self,
+                                                 id="QueryAPI",
+                                                 function=query_api_lambda,
+                                                 auth_type=lambda_.FunctionUrlAuthType.NONE,
+                                                 cors=lambda_.FunctionUrlCorsOptions(
+                                                                     allowed_origins=["*"],
+                                                                     allowed_methods=[lambda_.HttpMethod.GET, lambda_.HttpMethod.POST]))
```

Comment:
If it is only a query API, do we want to allow POST or only GET?

Reply:
removed POST